src/generator.js

```diff
@@ -17,6 +17,18 @@ function sanitizeText (text) {
   })
 }
 
+// Function to truncate long text and add a click-to-expand feature
+function truncateText (text, limit = 50) {
+  if (!text || text.length <= limit) return sanitizeText(text)
+
+  const truncated = sanitizeText(text.slice(0, limit))
+  const fullText = sanitizeText(text)
+
+  return `
```

Comment:
Instead we should probably use the details and summary elements so this can work without JS.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

Reply:
![Screen Shot 2025-02-27 at 11 08 54 AM](https://github.com/user-attachments/assets/443b2dbd-a526-4aa5-960d-fb723ea5bd26)
The text is appearing on a new line instead of staying inline, It's because the `<summary>` element is rendered as a block-level element by default.

Reply:
Does changing it to `inline` in CSS help?

Reply:
Nope `<summary class="expand" style="display: inline;"></summary>`

We can try `!important` to override?

Reply:
Hmm, feel free to just keep the JS stuff then, no need to fiddle too hard with this

Reply:
Okay